src/db/api/abstract.ts

```diff
@@ -8,9 +8,11 @@ export interface DatabaseApiOptions {
 export abstract class DatabaseService implements Service{
 
   protected db: IDatabaseController;
+  protected cache: Record<any, any>;
```

Comment:
I would rather use some cache with automatic item expiration or memory could grow larger than db. Also, I don't think you are using it :)

Reply:
whoops, I'll remove this. I was playing with an implementation of the db service that caches to get a feel for how the new interface could work